k2/csrc/rmepsilon.cc

```diff
@@ -224,19 +224,21 @@ void EpsilonsRemover<TracebackState>::GetSizes(
       int32_t arc_end = fsa.indexes[state + 1];
       for (int32_t arc_index = fsa.indexes[state]; arc_index != arc_end;
            ++arc_index) {
-        int32_t next_state = arcs_in[arc_index].dest_state;
-        int32_t label = arcs_in[arc_index].label;
-        float curr_arc_weight = arc_weights_in[arc_index];
+        const int32_t curr_arc_index = arc_index - fsa.indexes[0];
+        int32_t next_state = arcs_in[curr_arc_index].dest_state;
+        int32_t label = arcs_in[curr_arc_index].label;
+        float curr_arc_weight = arc_weights_in[curr_arc_index];
```

Comment:
index in `arc_weights_in` or `arc_map` should start from from `0`, so we need to minus `fsa.index[0]` in those indexes. Same changes in `determinize/weights/connect..`